examples/c/ex_col_store.c

```diff
@@ -0,0 +1,543 @@
+        error_check(cursor->get_key(cursor, &recno));
+        error_check(cursor->get_value(cursor, &loc_lat, &loc_long, &country));
+        if (strcmp("AUS", country) == 0) {
+            printf("Removing %s\n", country);
+            error_check(cursor->remove(cursor));
+        }
+    }
+    scan_end_check(ret == WT_NOTFOUND);
+    error_check(cursor->close(cursor));
+}
+
+static void
+generate_data(WEATHER *w_array)
+{
+    int country, day;
+
+    srand((unsigned int)getpid());
+
+    for (int i = 0; i < NUM_ENTRIES; i++) {
+        WEATHER w;
+        day = rand() % 7;
+        switch (day) {
+        case 0:
+            strcpy(w.day, "MON");
+            break;
+        case 1:
+            strcpy(w.day, "TUE");
+            break;
+        case 2:
+            strcpy(w.day, "WED");
+            break;
+        case 3:
+            strcpy(w.day, "THU");
+            break;
+        case 4:
+            strcpy(w.day, "FRI");
+            break;
+        case 5:
+            strcpy(w.day, "SAT");
+            break;
+        case 6:
+            strcpy(w.day, "SUN");
+            break;
+        default:
+            assert(false);
+        }
+        /* 24-hour-time 0-2400. */
+        w.hour = rand() % 2401;
+        /* Temperature range: 0-50C.  */
+        w.temp = rand() % 51;
+        /* Feels like temperature range 0-50C */
+        w.feels_like_temp = rand() % 51;
+        /* Humidity range: 0-100%. */
+        w.humidity = rand() % 101;
+        /* Pressure range: 900-1100pa */
+        w.pressure = (rand() % (1100 + 1 - 900)) + 900;
+        /* Wind range: 0-200 km/hr. */
+        w.wind = rand() % 201;
+        /* latitude: 0-180 degrees. */
+        w.loc_lat = rand() % 181;
+        /* longitude: 0-90 degrees. */
+        w.loc_long = rand() % 91;
+
+        country = rand() % 7;
+        switch (country) {
+        case 0:
+            strcpy(w.country, "AUS");
+            break;
+        case 1:
+            strcpy(w.country, "GBR");
+            break;
+        case 2:
+            strcpy(w.country, "USA");
+            break;
+        case 3:
+            strcpy(w.country, "NZD");
+            break;
+        case 4:
+            strcpy(w.country, "IND");
+            break;
+        case 5:
+            strcpy(w.country, "CHI");
+            break;
+        case 6:
+            strcpy(w.country, "RUS");
+            break;
+        default:
+            assert(false);
+        }
+
+        w_array[i] = w;
+    }
+}
+
+/*
+ * The function returns 0 when a valid min/max temperature can be calculated given the time range.
+ * If no records are found it will return WT_NOTFOUND, otherwise the program will crash if an
+ * internal error is encountered.
+ */
+static int
+find_min_and_max_temp(
+  WT_SESSION *session, uint16_t start_time, uint16_t end_time, int *min_temp, int *max_temp)
+{
+    WT_CURSOR *end_time_cursor, *join_cursor, *start_time_cursor;
+    uint64_t recno;
+    int exact;
+    int ret;
```

Comment:
Use `WT_DECL_RET` instead. That assigns to 0 automatically so you can remove the assignment below too.